File: Top 100 Codes/04-Important Codes related to Arrays/07-Sort first half in ascending order and second half in descending/Sortfirsthalfinascendingorderandsecondhalfindescending.py
def printOrder(arr, n):
    # Sorting the array
    arr.sort()

    # Printing first half in ascending order
    i = 0
    while i < n // 2:  # Integer division for correct indexing
        print(arr[i], end=" ")
        i += 1

    # Printing second half in descending order
    j = n - 1
    while j >= n // 2:
        print(arr[j], end=" ")
        j -= 1

# Driver code
arr = [5, 4, 6, 2, 1, 3, 8, 9, 7]
n = len(arr)

printOrder(arr, n)

File: Top 100 Codes/04-Important Codes related to Arrays/07-Sort first half in ascending order and second half in descending/Sortfirsthalfinascendingorderandsecondhalfindescending.cpp
#include <iostream>
#include <algorithm>

using namespace std;

void customSort(int arr[], int n) {
    // Sort the first half in ascending order
    sort(arr, arr + n / 2);

    // Sort the second half in descending order
    sort(arr + n / 2, arr + n, greater<int>());
}

int main() {
    int arr[] = {7, 2, 9, 1, 5, 6, 3, 8, 4, 10};
    int n = sizeof(arr) / sizeof(arr[0]);

    customSort(arr, n);

    cout << "Sorted Array: ";
    for (int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }

    return 0;
}
